### PR TITLE
Turn `Store` into `Accumulate`

### LINKED ISSUE #116: Turn `Store` into `Accumulate`
This would match the `AccumulateGrad` node of `torch.autograd`, and would make the `optimizer.zero_grad()` in the examples be less useless.

This is not backward-compatible so it would require a changelog entry.

### PR DESCRIPTION
**Pull request by PierreQuinton**
- Fixes #116 

might be relevant:
https://stackoverflow.com/questions/55266154/pytorch-preferred-way-to-copy-a-tensor

TL;DR: We might have to use `t.detach().clone()` instead of `t.clone()` to remove the computation graph dependence between the new grad and the previous, we might also want to detach the tensor we sum. This makes the accumulation non-differentiable.

I think we are suppose to detach, in torch, the way to be able to differentiate a differentiation seems to be by using `torch.autograd.grad` with `create_graph` set to `True`. The function `autograd.backward` or `tensor.backward` will set `.grad` fields that do not `require_grad`, they are therefore not in any graph.

### CODE REVIEW

File: src/torchjd/autojac/_transform/accumulate.py [hunk 1]
```diff
@@ -1,23 +1,27 @@
...
+        Accumulates gradients with respect to keys in their ``.grad`` field.
         """
 
         for key in gradients.keys():
             _check_expects_grad(key)
-            key.grad = gradients[key]
+            if hasattr(key, "grad") and key.grad is not None:
+                key.grad += gradients[key]
+            else:
+                key.grad = torch.clone(gradients[key])
```
Comment (PierreQuinton) [inline src/torchjd/autojac/_transform/accumulate.py:24]: As discussed this should probably be changed to
```python
grad_copy = gradients[key].detach().clone
if hasattr(key, "grad") and key.grad is not None:
   key.grad += grad_copy
else:
   key.grad = grad_copy
```

**Review by ValerianRey — APPROVED** [review]
